### PR TITLE
Change list behavior to allow starting indexing from 0.

### LINKED ISSUE #12827: Allow to start list index from 0
## 📝 Provide a description of the improvement

Currently when setting [list start index](https://ckeditor.com/docs/ckeditor5/latest/features/lists/lists.html#demo-3), you cannot go below value `1`. However, some use cases require starting list numbering from value `0`.
  
It would be good to enhance this feature and allow starting lists from `0`.

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by mremiszewski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Feature (lists): Allow list indexing to start from 0. Closes #12827.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-list/src/documentlistproperties/documentliststartcommand.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default class DocumentListStartCommand extends Command {
 
 		model.change( writer => {
 			for ( const block of blocks ) {
-				writer.setAttribute( 'listStart', options.startIndex || 1, block );
+				writer.setAttribute( 'listStart', options.startIndex >= 0 ? options.startIndex : 1, block );
```
Comment (arkflpc) [inline packages/ckeditor5-list/src/documentlistproperties/documentliststartcommand.js:53]: `!== null` would be more readable IMO.

File: packages/ckeditor5-list/src/documentlistproperties/documentlistpropertiesediting.js [hunk 2]
```diff
@@ -361,15 +361,16 @@ function createAttributeStrategies( enabledProperties ) {
...
 					writer.setAttribute( 'start', listStart, element );
 				} else {
 					writer.removeAttribute( 'start', element );
 				}
 			},
 
 			getAttributeOnUpcast( listParent ) {
-				return listParent.getAttribute( 'start' ) || 1;
+				const startAttributeValue = listParent.getAttribute( 'start' );
+				return startAttributeValue >= 0 ? startAttributeValue : 1;
```
Comment (niegowski) [inline packages/ckeditor5-list/src/documentlistproperties/documentlistpropertiesediting.js:373]:
```suggestion
				const startAttributeValue = listParent.getAttribute( 'start' );

				return startAttributeValue >= 0 ? startAttributeValue : 1;
```

File: packages/ckeditor5-list/tests/listproperties/liststartcommand.js [hunk 3]
```diff
@@ -372,5 +372,29 @@ describe( 'ListStartCommand', () => {
...
+			);
+
+			listStartCommand.execute( { startIndex: 0 } );
+
+			expect( getData( model ) ).to.equal(
+				'<listItem listIndent="0" listStart="0" listType="numbered">1.[]</listItem>'
+			);
+		} );
+
+		// eslint-disable-next-line mocha/no-exclusive-tests
```
Comment (niegowski) [inline packages/ckeditor5-list/tests/listproperties/liststartcommand.js:388]:
```suggestion
```

File: packages/ckeditor5-list/src/documentlistproperties/documentlistpropertiesediting.js [hunk 4]
```diff
@@ -361,15 +361,16 @@ function createAttributeStrategies( enabledProperties ) {
 			},
 
 			setAttributeOnDowncast( writer, listStart, element ) {
-				if ( listStart && listStart > 1 ) {
+				if ( listStart !== null && ( listStart === 0 || listStart > 1 ) ) {
```
Comment (niegowski) [inline packages/ckeditor5-list/src/documentlistproperties/documentlistpropertiesediting.js:364]:
```suggestion
				if ( listStart === 0 || listStart > 1 ) {
```

File: packages/ckeditor5-list/tests/documentlistproperties/documentlistpropertiesediting.js [hunk 5]
```diff
@@ -932,5 +932,68 @@ describe( 'DocumentListPropertiesEditing', () => {
 				` ) );
 			} );
 		} );
+
+		describe( 'upcasting', () => {
```
Comment (niegowski) [inline packages/ckeditor5-list/tests/documentlistproperties/documentlistpropertiesediting.js:936]: Please move conversion tests to `packages/ckeditor5-list/tests/documentlistproperties/converters.js`.

**Review by niegowski — APPROVED** [review]

**Pull request merged**